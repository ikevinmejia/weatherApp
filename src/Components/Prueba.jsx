import React, { useState } from 'react'

const Prueba = ({initialValue}) => {

  const [counter, setCounter] = useState(initialValue);

  const sumar = () => {
    setCounter(prev => prev + 1)
  }
  const restar = () => {
    setCounter(prev => prev - 1)
  }

  const reset = () => {
    setCounter(0)
  }

  return (
    <div>

      <h1>Contador</h1>
      <h2>La cuenta va en</h2>
      <h3>{counter}</h3>
      <button onClick={restar}>Restar</button>
      <button onClick={reset}>Reset</button>
      <button onClick={sumar}>Sumar</button>

    </div>
  )
}

export default Prueba