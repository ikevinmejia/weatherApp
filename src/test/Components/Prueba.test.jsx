import { render, screen, fireEvent } from "@testing-library/react"
import Prueba from "../../Components/Prueba"

describe('Test in <Prueba/>', () => {

    const value = 100

    test('match to snapshot', () => {

        const {container} = render(<Prueba initialValue={value}/>)
        expect(container).toMatchSnapshot()
    })

    test('Restar -1 con el bot贸n "Restar"', () => {

        render(<Prueba initialValue={value}/>)

        fireEvent.click(screen.getByText('Restar'))

        expect(screen.getByText('99')).toBeTruthy()
    })

    test('No debe sumar +1 con el bot贸n "Restar"', () => {

        render(<Prueba initialValue={value}/>)

        fireEvent.click(screen.getByText('Restar'))

        expect(screen.getByText('99')).not.toBeFalsy()
    })

    test('Sumar +1 con el bot贸n "Sumar"', () => {

        render(<Prueba initialValue={value}/>)

        fireEvent.click(screen.getByText('Sumar'))

        expect(screen.getByText('101')).toBeTruthy()
    })

    test('No debe restar -1 con el bot贸n "Sumar"', () => {

        render(<Prueba initialValue={value}/>)

        fireEvent.click(screen.getByText('Sumar'))

        expect(screen.getByText('101')).not.toBeFalsy()
    })


})