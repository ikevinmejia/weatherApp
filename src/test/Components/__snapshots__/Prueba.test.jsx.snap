// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Prueba/> match to snapshot 1`] = `
<div>
  <div>
    <h1>
      Contador
    </h1>
    <h2>
      La cuenta va en
    </h2>
    <h3>
      100
    </h3>
    <button>
      Restar
    </button>
    <button>
      Reset
    </button>
    <button>
      Sumar
    </button>
  </div>
</div>
`;
